src/app.py

```diff
@@ -13,39 +29,114 @@ def set_log_level(verbose):
 		logging.basicConfig(level=logging.DEBUG, format=log_format)
 
 def main():
-	parser = argparse.ArgumentParser(prog='oauth2-sec-test')
-	parser.add_argument('-c', '--client-id', required=True, help='OAuth2.0 Client ID')
-	parser.add_argument('-r', '--response-type', default='token', 
+    global global_endpoint, global_client_id, global_scope, global_response_type, global_redirect_uri
```

Comment:
Never use global variable. Any use of a global variable is an instant PR reject.